### ISSUE TITLE
javascript:S2589 false positive for array && array.some()

### ISSUE THREAD

**Issue by imranariffin** [SonarSource/SonarJS#1043]
**I want to report a bug.**
Sonar complains that the following reproducer code will always return true but that is wrong. See example to prove this in reproducer code below.

**SonarQube version:** Version 6.7.3<br>

**Rule key:** javascript:S2589

**Reproducer** 
```
// minimal reproducer if relevant
const pets = getPets(); // returns array of objects e.g. [{type: 'CAT'}, {type:'DOG'}]
const hasDog = pets && pets.some(p => p.type === 'DOG') // MAJOR - Refactor this code so that this expression does not always evaluate to true. [javascript:S2589]

// Example that will evaluate hasDog to false:
const hasDog = [] && [].some(p => p.type === 'DOG');  // => false
```

**Expected behavior**
No Issue

**Comment by imranariffin** [comment]
Closing. I misunderstood the issue. There was an if statement before the line that looks like this:
```
if (!pets) {
    return;
}
```
so it was complaining because ```hasDog = pets.some(p => p.type === 'DOG')``` should be sufficient.
